### PR TITLE
Add proper Create signature

### PR DESCRIPTION
**Pull request by marcushines**
Add coverage for Update

### CODE REVIEW

File: api/clientset/v1beta1/topology_test.go [hunk 1]
```diff
@@ -78,7 +80,26 @@ func setUp(t *testing.T) (*Clientset, *restfake.RESTClient) {
...
 	cs.restClient = fakeClient
-	cs.dInterface = dynamicfake.NewSimpleDynamicClient(scheme.Scheme, objs...).Resource(gvr)
+	f := dynamicfake.NewSimpleDynamicClient(scheme.Scheme, objs...)
+	// Add handler for Update call.
+	f.PrependReactor("update", "*", func(action ktest.Action) (bool, runtime.Object, error) {
+		uAction, ok := action.(ktest.UpdateAction)
+		if !ok {
+			return false, nil, nil
+		}
+		fmt.Println(action)
```
Comment (alexmasi) [inline api/clientset/v1beta1/topology_test.go:90]: Add context to what is being printed for looking at the test logs

(here and with tObj)

File: api/clientset/v1beta1/topology_test.go [hunk 2]
```diff
@@ -111,7 +132,7 @@ func TestCreate(t *testing.T) {
 		}
 		t.Run(tt.desc, func(t *testing.T) {
 			tc := cs.Topology("foo")
-			got, err := tc.Create(context.Background(), tt.want)
+			got, err := tc.Create(context.Background(), tt.want, metav1.CreateOptions{})
```
Comment (alexmasi) [inline api/clientset/v1beta1/topology_test.go:135]: Would a test case that sets one of these options be valuable here?

File: api/clientset/v1beta1/topology_test.go [hunk 3]
```diff
@@ -289,11 +310,59 @@ func TestWatch(t *testing.T) {
...
+				t.Fatalf("failed to generate update: %v", err)
+			}
+			got, err := tc.Update(context.Background(), &unstructured.Unstructured{Object: update}, metav1.UpdateOptions{})
+			if s := errdiff.Substring(err, tt.wantErr); s != "" {
+				t.Fatalf("unexpected error: %s", s)
+			}
+			if tt.wantErr != "" {
+				return
+			}
+			want := tt.want.DeepCopy()
```
Comment (alexmasi) [inline api/clientset/v1beta1/topology_test.go:353]: why do you need a deep copy and comparison with reflection here instead of cmp/opts?

**Review by alexmasi — APPROVED** [review]

File: api/clientset/v1beta1/topology_test.go (hunk 2 above)
Comment (marcushines) [reply api/clientset/v1beta1/topology_test.go:133]: nope you never set these in any general case

File: api/clientset/v1beta1/topology_test.go (hunk 3 above)
Comment (marcushines) [reply api/clientset/v1beta1/topology_test.go:351]: cause cmp wasn't widely used when first did this i will flip it at some point

**Review by alexmasi — APPROVED** [review]

File: api/clientset/v1beta1/topology_test.go (hunk 3 above)
Comment (hellt) [reply api/clientset/v1beta1/topology_test.go:351]: if you won't do a deepcopy, won't you change the `obj` vars in-place as is is a pointer, and risk mutating an object

**Pull request merged**